### PR TITLE
Add tests for comments in more expression-like contexts

### PR DESCRIPTION
**Pull request by nex3**
See sass/dart-sass#2263

[skip dart-sass]

### CODE REVIEW

**Comment by ntkme** [comment]
I wonder if we should omit the whitespace in between silent comment and sibling tokens during output?

**Comment by nex3** [comment]
It's not generally safe to omit that whitespace entirely—for example, `foo//\nbar` is very different than `foobar`. There are some cases where we could fold more aggressively than we do now, but that's generally true with newlines adjacent to trailing spaces, not just silent comments.

**Review by Goodwine — APPROVED** [review]

File: spec/css/functions/special.hrx [hunk 1]
```diff
@@ -25,3 +25,187 @@ a {b: -a-calc(#{0})}
...
+<===> comment/progid/before_close_paren/loud/output.css
+a {
+  b: progid:c(d /**/);
+}
+
+<===>
+================================================================================
+<===> comment/progid/before_close_paren/silent/input.scss
+a {
+  b: progid:c(d //
```
Comment (Goodwine) [inline spec/css/functions/special.hrx:204]: If there are multiple lines with `//`, are they considered a single silent comment, or multiple silent comments?

For example, would this block:
```scss
a {
  b: progid:c(d // foo
                // bar
    );
}
```
result in this?:
```css
a {
  b: progid:c(d

    );
}
```
And if so, is that OK?

Comment (nex3) [reply spec/css/functions/special.hrx:204]: They are not considered a single comment, but the whitespace around them is still considered "adjacent" so you won't end up with multiple newlines.

**Comment by ntkme** [comment]
> There are some cases where we could fold more aggressively than we do now

If we treat silent comment as a kind of whitespace when processing AST, I think this can potentially allow us to fold it more aggressively based on context, even under the current logic.

For example, if silent comment is treated as whitespace: 
`a { b: calc(c<whitespace><silent-comment><newline><whitespace>) }` can be parsed as `a { b: calc(c<whitespace>) }`, and this whitespace can be completely striped based on its context.

Not sure if there is any corner case, and I haven't looked at the implementation details so not sure how much work it is. If it's too much work we can just leave it as is.

**Pull request merged**